service/resource/configmap/update_test.go

```diff
@@ -1 +1,426 @@
+		{
+			currentState: nil,
+			desiredState: nil,
+
+			expectedUpdateStateConfigMap: nil,
+			expectedErrorHandler:         nil,
+		},
+
+		// Test that that if the current state and desired state are the same,
+		// and the configmap is empty,
+		// the update state is nil.
+		{
+			currentState: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: configMapNamespace,
+				},
+				Data: map[string]string{},
+			},
+			desiredState: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: configMapNamespace,
+				},
+				Data: map[string]string{},
+			},
+
+			expectedUpdateStateConfigMap: nil,
+			expectedErrorHandler:         nil,
+		},
+
+		// Test that if the current state and desired state are the same,
+		// the update state is nil.
+		{
+			currentState: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: configMapNamespace,
+				},
+				Data: map[string]string{
+					configMapKey: "foo",
+				},
+			},
+			desiredState: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: configMapNamespace,
+				},
+				Data: map[string]string{
+					configMapKey: "foo",
+				},
+			},
+
+			expectedUpdateStateConfigMap: nil,
+			expectedErrorHandler:         nil,
+		},
+
+		// Test that if the current state and desired state are different,
+		// the update state matches the desired state.
+		{
+			currentState: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: configMapNamespace,
+				},
+				Data: map[string]string{
+					configMapKey: "foo",
+				},
+			},
+			desiredState: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: configMapNamespace,
+				},
+				Data: map[string]string{
+					configMapKey: "bar",
+				},
+			},
+
+			expectedUpdateStateConfigMap: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: configMapNamespace,
+				},
+				Data: map[string]string{
+					configMapKey: "bar",
+				},
+			},
+			expectedErrorHandler: nil,
+		},
+
+		// Test that if the current state and desired state configmaps have
+		// different names, an error is returned.
+		{
+			currentState: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: configMapNamespace,
+				},
+				Data: map[string]string{
+					configMapKey: "foo",
+				},
+			},
+			desiredState: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "another-name",
+					Namespace: configMapNamespace,
+				},
+				Data: map[string]string{
+					configMapKey: "foo",
+				},
+			},
+
+			expectedUpdateStateConfigMap: nil,
+			expectedErrorHandler:         IsConfigMapWrongName,
+		},
+
+		// Test that if the current state and desired state configmaps are in
+		// different namespaces, an error is returned.
```

Comment:
Why should that ever happen?

Reply:
Misconfiguration or some such. Being defensive.